R/zzz.R

```diff
@@ -1,5 +1,27 @@
 .lockbox_env <- new.env()
 
+.get_safely_from_envir <- function(target, default = NULL) {
+  res <- tryCatch(
+    get(target, envir = .lockbox_env)
+  , error = function(e) {default})
+}
+
+set_session_id <- function() {
+  if (is.null(.get_safely_from_envir("session_id"))) {
+    if (requireNamespace("uuid", quietly = TRUE)) {
+      assign("session_id", uuid::UUIDgenerate(), envir = .lockbox_env)
+    } else {
+      assign("session_id", digest::digest(as.integer(Sys.time())), envir = .lockbox_env)
+    }
+  }
+  
+  # We don't want to clutter up ~/.R/ with tons of symlinked dirs, lets try to do some housekeeping
+  reg.finalizer(.lockbox_env, function(env) {
```

Comment:
This is my first time working with the finalizers.  Based on my reading so far, it is getting called because something tried to destroy `.lockbox_env`.  I was/am not certain how much of `package:lockbox` will be callable at that point.  I'd have to do that testing tomorrow.